test/e2e/argo-workflows/agent.yaml

```diff
@@ -322,6 +322,181 @@ spec:
+          - secrets
+          - nodes
+          - pods
+          - services
+          - resourcequotas
+          - replicationcontrollers
+          - limitranges
+          - persistentvolumeclaims
+          - persistentvolumes
+          - namespaces
+          - endpoints
+          verbs: ["list", "watch"]
+        - apiGroups: ["extensions"]
+          resources:
+          - daemonsets
+          - deployments
+          - replicasets
+          verbs: ["list", "watch"]
+        - apiGroups: ["apps"]
+          resources:
+          - statefulsets
+          verbs: ["list", "watch"]
+        - apiGroups: ["batch"]
+          resources:
+          - cronjobs
+          - jobs
+          verbs: ["list", "watch"]
+        - apiGroups: ["autoscaling"]
+          resources:
+          - horizontalpodautoscalers
+          verbs: ["list", "watch"]
+        ---
+        apiVersion: rbac.authorization.k8s.io/v1
+        kind: RoleBinding
+        metadata:
+          name: kube-state-metrics
+          namespace: kube-system
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: Role
+          name: kube-state-metrics-resizer
+        subjects:
+        - kind: ServiceAccount
+          name: kube-state-metrics
+          namespace: kube-system
+        ---
+        apiVersion: rbac.authorization.k8s.io/v1
+        kind: Role
+        metadata:
+          namespace: kube-system
+          name: kube-state-metrics-resizer
+        rules:
+        - apiGroups: [""]
+          resources:
+          - pods
+          verbs: ["get"]
+        - apiGroups: ["extensions"]
+          resources:
+          - deployments
+          resourceNames: ["kube-state-metrics"]
+          verbs: ["get", "update"]
+        ---
+        apiVersion: v1
+        kind: ServiceAccount
+        metadata:
+          name: kube-state-metrics
+          namespace: kube-system
+
+    - name: kube-state-metrics-deployment
+      value: |
+        apiVersion: apps/v1beta2
+        kind: Deployment
+        metadata:
+          name: kube-state-metrics
+          namespace: kube-system
+        spec:
+          selector:
+            matchLabels:
+              k8s-app: kube-state-metrics
+          replicas: 1
+          template:
+            metadata:
+              labels:
+                k8s-app: kube-state-metrics
+            spec:
+              serviceAccountName: kube-state-metrics
+              containers:
+              - name: kube-state-metrics
+                image: quay.io/coreos/kube-state-metrics:v1.3.1
+                ports:
+                - name: http-metrics
+                  containerPort: 8080
+                - name: telemetry
+                  containerPort: 8081
+                readinessProbe:
+                  httpGet:
+                    path: /healthz
+                    port: 8080
+                  initialDelaySeconds: 5
+                  timeoutSeconds: 5
+              - name: addon-resizer
```

Comment:
Do we need the addon-resizer?

Reply:
Maybe not. Let me see if it still works without that.